generators/app/templates/docker-compose.yml

```diff
@@ -5,8 +5,8 @@ services:
     image: <%= imageName %><% if (environment === 'debug') { %>:debug<% } %>
     build:
       context: .
-      dockerfile: Dockerfile<% if (environment === 'debug') { %>.debug<% } %><% if (isWebProject && portNumber) { %>
-    ports:
+      dockerfile: Dockerfile<% if (environment === 'debug') { %>.debug<% } %><% if ((isWebProject && portNumber) || (environment === 'debug' && debugPortNumber)) { %>
```

Comment:
The below (environment === 'debug') has a broader block containing check for volumeMap and includeComposeForDebug along with debugPortNumber.

---
In reply to: [71240674](https://github.com/Microsoft/generator-docker/pull/103#discussion_r71240674) [](ancestors = 71240674)